packages/mermaid/src/diagrams/git/gitGraphRenderer.js

```diff
@@ -275,32 +275,36 @@ const drawCommits = (svg, commits, modifyGraph) => {
         }
       }
       if (commit.tag) {
-        const rect = gLabels.insert('polygon');
-        const hole = gLabels.append('circle');
-        const tag = gLabels
+        const g = gLabels.append('g');
+        if (gitGraphConfig.rotateTagLabel && dir === 'LR') {
+          g.attr('transform', 'translate(27,1) rotate(-45, ' + pos + ',' + y + ')');
+        }
+        const rect = g.insert('polygon');
+        const hole = g.append('circle');
+        const tag = g
           .append('text')
           // Note that we are delaying setting the x position until we know the width of the text
           .attr('y', y - 16)
           .attr('class', 'tag-label')
           .text(commit.tag);
         let tagBbox = tag.node().getBBox();
-        tag.attr('x', pos + 10 - tagBbox.width / 2);
+        tag.attr('x', pos + 10);
 
         const h2 = tagBbox.height / 2;
         const ly = y - 19.2;
         rect.attr('class', 'tag-label-bkg').attr(
           'points',
           `
-          ${pos - tagBbox.width / 2 - px / 2},${ly + py}
-          ${pos - tagBbox.width / 2 - px / 2},${ly - py}
-          ${pos + 10 - tagBbox.width / 2 - px},${ly - h2 - py}
-          ${pos + 10 + tagBbox.width / 2 + px},${ly - h2 - py}
-          ${pos + 10 + tagBbox.width / 2 + px},${ly + h2 + py}
-          ${pos + 10 - tagBbox.width / 2 - px},${ly + h2 + py}`
+          ${pos - px / 2},${ly + py}
+          ${pos - px / 2},${ly - py}
+          ${pos + 10 - px},${ly - h2 - py}
+          ${pos + 10 + tagBbox.width + px},${ly - h2 - py}
+          ${pos + 10 + tagBbox.width + px},${ly + h2 + py}
+          ${pos + 10 - px},${ly + h2 + py}`
```

Comment:
Warning: This will conflict with https://github.com/mermaid-js/mermaid/pull/5161
One of you will have to sync based on which is merged first :)

Reply:
I removed these changes because it was related to another feature (turning the tags).
I will probably open another PR for that.